addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -18,6 +18,15 @@ export class DiscussCoreCommon {
     }
 
     setup() {
+        this.busService.addEventListener(
+            "connect",
+            () =>
+                this.store.imStatusTrackedPersonas.forEach((p) => {
+                    const model = p.type === "partner" ? "res.partner" : "mail.guest";
+                    this.busService.addChannel(`odoo-presence-${model}_${p.id}`);
+                }),
+            { once: true }
```

Comment:
Is that handling re-connects correctly with once?
If `imStatusTrackedPersonas` changes during `isActive` downtime, is that possible?

Reply:
I don't think so. `isActive` switches to `true` when the worker is first started. It never goes back afterwards.